data/src/main/java/com/linkedin/data/ByteString.java

```diff
@@ -481,6 +481,26 @@ public int feed(ByteArrayFeeder feeder, int index) throws IOException
     return returnIndex < _byteArrays.getArraySize() ? returnIndex : -1;
   }
 
+  /**
+   * Feeds a chunk of this {@link ByteString} to a {@link com.linkedin.data.DataParser}
+   * without copying the underlying byte[].
+   *
+   * @param parser the feeder to feed the bytes to
+   * @param index the index of the chunk to feed
+   *
+   * @throws IOException if an error occurs while writing to the feeder
+   *
+   * @return The next index to feed or -1 if no more indices are left to feed.
+   */
+  public int feed(DataParser parser, int index) throws IOException
+  {
+    ByteArray byteArray = _byteArrays.get(index);
+    int end = byteArray.getOffset() + byteArray.getLength();
+    parser.feedInput(byteArray.getArray(), byteArray.getOffset(), end);
```

Comment:
Typically APIs take offset and length. The jackson API is kinda unique in the sense that it takes end. How about we make the API contract take offset, length and modify the jackson impl to pass in end internally?